araproc/analysis/standard_reco.py

```diff
@@ -701,3 +704,135 @@ def min_frac_corr_depth_multiple(self, *maps, fraction=0.6, z_thresh=0):
             raise RuntimeError("No maps meet the fractional correlation threshold within the z_thresh.")
 
         return min_depth, min_depth_index
+
+    def pairing(channels):
+
+        """
+        Makes channel pairs.
+    
+        Parameters
+        ----------
+        channels: list
+           List of given channels.
+
+        Returns
+        -------
+        pairs: list
+           List of all possible pair combinations of given channels.
+        """
+
+        pairs = list(itertools.combinations(channels, 2))
+
+        return pairs
+
+    def get_corr(y1, y2):
+
+        """
+        Calculates normalized correlation between two waveforms, y1 and y2.
+    
+        Parameters
+        ----------
+        y1: list
+           List of values of waveform 1.
+        y2: list
+           List of values of waveform 2.
+    
+        Returns
+        -------
+        corr: array
+           Array of values of cross correlation function.  
+        """
+
+        _, y1 = wfu.tgraph_to_arrays(y1)
+        _, y2 = wfu.tgraph_to_arrays(y2)
+
+        n = len(y1)
+        corr = co(y2, y1, mode = 'same')/np.sqrt(co(y1, y1, mode = 'same')[int(n/2)]*co(y2, y2, mode = 'same')[int(n/2)])
+
+        return corr
+
+    def get_corr_snr(first_ch, second_ch):
+
+        """
+        Calculates channel-pair correlation SNR
+    
+        Parameters
+        ----------
+        first_ch: array 
+           Array of a given channel waveform.
+        second_ch: array 
+           Array of another given channel waveform.
+     
+        Returns
+        -------
+        corr_snr: float
+           Channel pair correlation SNR.
+        """
+
+        corr = get_corr(first_ch, second_ch)
+        corr_snr = snr.get_snr(corr)
+
+        return corr_snr
+
+    def get_avg_corr_snr(wave_bundle, excluded_channels = []):
```

Comment:
Same here, will need a `self` argument. Also will need to take an `event_number` argument for technical reasons.